rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,160 @@
                 <matches pattern="(TableGenerator|TableHiLoGenerator|SequenceGenerator|SequenceIdentityGenerator|SequenceHiLoGenerator)" />
             </where>
         </rule>
+        <rule id="hibernate4-00029">
+            <when>
+                <javaclass references="org.hibernate.type.descriptor.sql.SqlTypeDescriptor">
+<!--                     <location>INHERITANCE</location> -->
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Class org.hibernate.type.descriptor.sql.SqlTypeDescriptor change in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Built-in `org.hibernate.type.descriptor.sql.SqlTypeDescriptor` implementations no longer auto-register themselves with `org.hibernate.type.descriptor.sql.SqlTypeDescriptorRegistry`.
+                     Applications using custom `SqlTypeDescriptor` implementations extending the built-in ones and relying on that behavior
+                      should be updated to call `SqlTypeDescriptorRegistry#addDescriptor` themselves.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#type-handling"
+                        title="Hibernate 5 Type Handling" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00030">
+            <when>
+                <javaclass references="org.hibernate.hql.spi.{type}" />
+            </when>
+            <perform>
+                <hint title="Class org.hibernate.hql.spi.{type} were moved in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Hibernate 5 moved `org.hibernate.hql.spi.{type}` class to new `org.hibernate.hql.spi.id` package.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#misc"
+                        title="Hibernate 5 Misc changes" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+            <where param="type">
+                <matches pattern="(MultiTableBulkIdStrategy|MultiTableBulkIdStrategy.DeleteHandler|MultiTableBulkIdStrategy.UpdateHandler|AbstractTableBasedBulkIdHandler|AbstractTableBasedBulkIdHandler.ProcessedWhereClause|TableBasedDeleteHandlerImpl|TableBasedUpdateHandlerImpl)" />
+            </where>
+        </rule>
+        <rule id="hibernate4-00031">
+            <when>
+                <javaclass references="org.hibernate.hql.spi.PersistentTableBulkIdStrategy" />
```

Comment:
This class can also be referenced in configuration under key `hibernate.hql.bulk_id_strategy`. Should have a rule for that?